### ISSUE TITLE
NVDA not announcing british currency correctly when number processing is enabled

### ISSUE THREAD

**Issue by mobeeniqbal** [ABuffEr/numberProcessing#10]
Hi.
This is a fantastic add-on, many thanks for all your hard work. I have encountered a small bug which I am hoping can be rectified. I have noticed with the latest update that NVDA does not read financial transactions correctly, for example when looking at a bank statement. Here is an example.
I have the minimum number of digits set to 5.
If I type £3000, NVDA reads this as 3 thousand pounds which is what I would expect.
However if I type £30000, NVDA reads this as 3 0 0 0 0 pounds. So the pounds symbol is read after the amount, and the currency amount is not properly recognised. What I would expect it to say is thirty thousand pounds.
Let's take another example. If the amount is £3,650, NVDA reads this as three, six thousand 500 pounds. When the add-on is disabled, NVDA reads the currency amounts correctly. This also seems to affect transactions starting with the $ symbol, I have not tested euros.
Is there any way of implementing a function to specify which symbols should be excluded from processing? Or including a list of currency symbols that the add-on will ignore by default? as it seems to also affect other currencies. So if the number has a £ symbol at the start it will not be processed at all regardless of the number of digits setting. in general. I am using Windows 10 with NVDA 2024.3.1 with version 1.0.20240820 of the add-on. I hope this all makes sense? I'd be happy to test any changes to the add-on and provide feedback.
Kind regards,
Mo.

**Comment by ABuffEr** [comment]
Hi,
sorry for delay. The trailing read of currency symbol is intentional, because some synthesizers misunderstand it receiving it followed by a around-spaced digit (what the add-on does); e.g.: £30000 would be processed as:
£  3  0  0  0  0
and read as: three £ zero zero etc.
Note that to hear thirty thousands you have to disable the add-on, or set process length to 6.
Anyway, at this point I'll introduce an option to adjust the mechanism, according to synthesizer behavior.

**Comment by mobeeniqbal** [comment]
Hi.
Many thanks for confirming the issue and how to resolve it. I did test the behaviour with ESpeak and Eloquence and they both behaved the same. I look forward to the option to adjust the mechanism being implemented. I'd be happy to carry out further testing if required.

**Comment by NameNoQuality** [comment]
d0991c91836a7cf37cc051aa4be3d19a6d0640ea9a152c535764f62fe51e7ddf  Strategy 2021-2025.pdf

**Comment by NameNoQuality** [comment]
> #3 
< #9 {d0991c91836a7cf37cc051aa4be3d19a6d0640ea9a152c535764f62fe51e7ddf Strategy 2021-2025.pdf}
